Build bug - Makefile:40: install Error 1

Running `make install` on a clean system can produce this error -:
```
$ make install

...
rm -f ~/go/bin/polycli
cp ./out/polycli ~/go/bin/
cp: cannot create regular file '/home/user/go/bin/': Not a directory
make: *** [Makefile:40: install] Error 1
```

The INSTALL_DIR  (~/go/bin/) needs to get created first.

--- a/Makefile
+++ b/Makefile
@@ -37,6 +37,7 @@ build: $(BUILD_DIR) ## Build go binary.
 .PHONY: install
 install: build ## Install the go binary.
 	$(RM) $(INSTALL_DIR)/$(BIN_NAME)
+	mkdir -p $(INSTALL_DIR)
 	cp $(BUILD_DIR)/$(BIN_NAME) $(INSTALL_DIR)/
 
 .PHONY: cross

